src/script/components/RichTextEditor/plugins/MentionsPlugin/MentionsPlugin.tsx

```diff
@@ -162,10 +161,11 @@ export function MentionsPlugin({onSearch, openStateRef}: MentionsPluginProps) {
   return (
     <TypeaheadMenuPlugin
       onQueryChange={setQueryString}
-      onSelectOption={handleSelectOption}
+      onSelectOption={insertMention}
       triggerFn={checkForMentionMatch}
       options={options}
       menuRenderFn={menuRenderFn}
+      onClose={() => (openStateRef.current = false)}
```

Comment:
This is the actual change. The rest is mostly cleaning up of unneeded operations